tests/c_benchmarks/bm_tcp_adapter.cpp

```diff
@@ -103,3 +221,143 @@ static void BM_TCPEchoServerLatencyWithoutQDR(benchmark::State &state)
 }
 
 BENCHMARK(BM_TCPEchoServerLatencyWithoutQDR)->Unit(benchmark::kMillisecond);
+
+class DispatchRouterThreadTCPLatencyTest
+{
+    QDR mQdr{};
+    std::thread mT;
+
+   public:
+    DispatchRouterThreadTCPLatencyTest(const std::string configName)
+    {
+        Latch mx;
+
+        mT = std::thread([&mx, this, &configName]() {
+            mQdr.initialize(configName);
+            mQdr.wait();
+
+            mx.notify();
+            mQdr.run();
+
+            mQdr.deinitialize(false);
+        });
+
+        mx.wait();
+    }
+
+    ~DispatchRouterThreadTCPLatencyTest()
+    {
+        mQdr.stop();
+        mT.join();
+    }
+};
+
+class DispatchRouterSubprocessTcpLatencyTest
+{
+    int pid;
+
+   public:
+    DispatchRouterSubprocessTcpLatencyTest(std::string configName)
+    {
+        pid = fork();
```

Comment:
I really like this! Since dispatch is compiled in, it can be exec'd by forking the test/benchmark process. With (considerable) work, it would be possible to even have two-way interactions with the forked process, using some Linux IPC... Just running qdmanage is simpler and probably better, though.